v2/apierror/apierror.go

```diff
@@ -256,27 +256,22 @@ func (a *APIError) setDetailsFromError(err error) bool {
 
 // FromError parses a Status error or a googleapi.Error and builds an APIError.
 func FromError(err error) (*APIError, bool) {
-	if err == nil {
-		return nil, false
-	}
-
-	ae := APIError{err: err}
-	if !ae.setDetailsFromError(err) {
-		return nil, false
-	}
-	return &ae, true
+	return ParseError(err, true)
 }
 
-// FromWrappingError parses a Status error or a googleapi.Error and
-// builds an APIError, but to avoid a cycle when the provided error
-// will externally wrap the new APIError, it does not set err in the
-// new APIError.
-func FromWrappingError(err error) (*APIError, bool) {
+// ParseError parses a Status error or a googleapi.Error and
+// builds an APIError. If wrap is true, it sets err in the
```

Comment:
Maybe simplify the doc here, you are referencing internal bits in docs for a public api. Maybe just say something like if true APIError will wrap the provided error.

Reply:
Fixed, thanks!

Reply:
Thanks for simplifying, but we still have the same issue here of referencing an unexported fields in the public docs. Rather descibe the effect that is wraps the error instead of saying it sets a field.

Reply:
Updated this and `FromError` as well, PTAL.